src/mlpack/methods/ann/gan/gan.hpp

```diff
@@ -0,0 +1,357 @@
+/**
+ * @file gan.hpp
+ * @author Kris Singh
+ * @author Shikhar Jaiswal
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_GAN_GAN_HPP
+#define MLPACK_METHODS_ANN_GAN_GAN_HPP
+
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/gan/gan_policies.hpp>
+#include <mlpack/methods/ann/visitor/output_parameter_visitor.hpp>
+#include <mlpack/methods/ann/visitor/reset_visitor.hpp>
+#include <mlpack/methods/ann/visitor/weight_size_visitor.hpp>
+#include <mlpack/methods/ann/visitor/weight_set_visitor.hpp>
+
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. **/ {
+
+/**
+ * The implementation of the standard GAN module. Generative Adversarial
+ * Networks (GANs) are a class of artificial intelligence algorithms used
+ * in unsupervised machine learning, implemented by a system of two neural
+ * networks contesting with each other in a zero-sum game framework. This
+ * technique can generate photographs that look at least superficially
+ * authentic to human observers, having many realistic characteristics.
```

Comment:
It would be really useful to have some example usage here or in some tutorial somewhere.  I think just from the constructor I understand how I would use it, but examples are always great starting points for users.